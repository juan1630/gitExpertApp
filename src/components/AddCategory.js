import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {

// se debe de dejar elstrig vacio
  const [inputValue, setInputValue] =useState('');

  const handleValue = (e) => {

      setInputValue( e.target.value );
  }

  const handleSubmit = (e) => {
    // console.log(e );
    e.preventDefault();

    if( inputValue.trim().length > 2 ){
            // se previene que se recargue la pagina
      setCategories( cats => [inputValue, ...cats] );
      // recibimso por defecto los valores de la catgoria
      setInputValue('');
      }


  }

  return(

    // el fargment se quita si hay otro agrupador
    <form   onSubmit={ handleSubmit }  >

      <input type="text"
      className="input"
      value={inputValue}
      onChange={  handleValue }
      />
      </form>
  )
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
