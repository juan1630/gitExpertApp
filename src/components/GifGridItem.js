import React from 'react';
import PropTypes from 'prop-types'

export const GifGridItem = ( { url, title} ) => {

        // console.log( props );
        return ( <div className="card animate__animated animate__bounce animate__delay-5s"  >
          <img src={url} alt={title} />
          <p> { title } </p>
         </div>)
        }



GifGridItem.propTypes = {
  url:PropTypes.string.isRequired,
  title:PropTypes.string.isRequired,
}
