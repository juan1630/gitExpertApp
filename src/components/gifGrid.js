import React from 'react';
import { useFetchGif } from '../hooks/useFetchGif';
import {GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {


      const { data: images, loading }  =  useFetchGif( category );
      // console.log( data );
      // console.log( loading );

      { loading && <p>loading </p> }

        return (
          <div className="" >
            <h3> { category } </h3>
                <div className="card-grid" >
                  {
                    images.map( (img) =>(

                      <GifGridItem key={img.id} {...img} />
                    )
                    )
                  }
                </div>
            </div>)
        }

        // { loading ? 'Cargando...' : 'Fin de la carga' }
