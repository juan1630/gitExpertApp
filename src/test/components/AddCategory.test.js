import React from 'react';
import { shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en el componente AddCategory',  ()=> {


  const setCategories = jest.fn();

  let wrapper;

  beforeEach(()=> {
    jest.clearAllMocks();
    //limpia los mocks
    wrapper= shallow( <AddCategory  setCategories={setCategories} />  );
  })


    test('Debe de mostrar el componente', ()=> {

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', ()=> {
        const input = wrapper.find('input');
        // console.log( input );
        const value = "Algo";
        // mandamos el valor por medio de un json
        input.simulate('change', { target: { value } });

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( value)
    });


    test('No debe de hacer el post del formulario', ()=> {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // se ejectu la funcion
        expect( setCategories ).not.toHaveBeenCalled();

    });
});
