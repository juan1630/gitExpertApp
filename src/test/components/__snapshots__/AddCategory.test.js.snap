// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe de mostrar el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
